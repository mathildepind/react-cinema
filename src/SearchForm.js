import React from 'react';

class SearchForm extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      movieSearch : '',
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  handleChange(event){
    let searchInput = event.target.value;
    this.setState({
      movieSearch: searchInput
    });
    console.log(this.state.movieSearch);
  }


  handleClick(event){
    event.preventDefault();
    let searchInput = this.state.movieSearch;
    this.props.receiver(searchInput);
    console.log(searchInput);
  }


  render(){
    return (
      <div>
        <form onSubmit={this.handleClick}>
          <label>
            <input
              type="text"
              value={this.state.movieSearch}
              onChange={this.handleChange}
            />
          </label>
            <button type='submit'>Search</button>
        </form>

        {/* <button onClick={this.handleClick}>Search</button> */}
      </div>
    );
  }
};

export default SearchForm;
