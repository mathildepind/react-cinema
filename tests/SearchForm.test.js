import React from 'react';
import SearchForm from '../src/SearchForm';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('SearchForm component', () => {
  it('correctly renders tree',()=>{
    const tree = renderer.create(<SearchForm/>).toJSON();
    expect (tree).toMatchSnapshot();
  })

  it('starts with a movie', () =>{
    const wrapper = shallow(<SearchForm />);
    const event = {
      target: {
        value: 'batman'
      }
    }
    wrapper.find('input').simulate('change', event);
    // const text = wrapper.state('movieSearch');
    // expect(text).toEqual('batman');
    const value = wrapper.find('input').props().value;
    expect(value).toBe('batman');
  });

  it('calls reciever prop function with value state',()=>{
    const receiver = jest.fn();
    const event = {
      preventDefault: jest.fn()
    };

    const wrapper = shallow(<SearchForm receiver={receiver} />);
    wrapper.setState({movieSearch:'Aquaman'});
    wrapper.find('form').simulate('submit',event);

    expect(receiver.mock.calls).toEqual([
      ['Aquaman']
    ]);

    expect(event.preventDefault.mock.calls).toEqual([[]]);

});


});
