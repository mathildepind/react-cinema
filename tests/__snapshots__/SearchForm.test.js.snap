// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component correctly renders tree 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
